## Z-Test Score

In [4]:
from __future__ import print_function
__author__ = 'Franz Loza CUSP 2016'
import csv
import requests
import sys
import os

import numpy as np
from numpy.random import randn
import pylab as pl
import pandas as pd

In [2]:
data = !curl -O https://raw.githubusercontent.com/fedhere/PUI2016_fb55/master/Lab3_fb55/times.txt

In [5]:
#Given: ~N(u=36,o=6)
Told_mean = 36
Told_std = 6

busdata = pd.read_csv('times.txt', sep=" ", header = None, )
#busdata
Tnew_Mean = busdata.mean()
#busdata.describe()
Tnew_Mean

0    34.466162
dtype: float64

## Tnew or the new average commute time is 34.466162 which is less than the average of the old bus route.

In [5]:
#pl.hist(busdata)
pl.plot(busdata,".")
pl.show

#is the new route taking the same or less time than before
#take mean of old time and old time
#difference of two means comparable to these 2 standard deviations
#demominator 
#=z is the standard deviation away from the mean (when you are testing 2 means that are gaussian)
#reject the null hypothesis
# mean of the sample = xbar
# mean of sample and mean of 

#if number is greater than threshold --> reject null hypothesis

<function matplotlib.pyplot.show>

In [6]:
busdata.std()

0    7.13728
dtype: float64

In [7]:
z_score = (Tnew_Mean-Told_mean)/(Told_std/np.sqrt(len(busdata)))
z_score

0   -2.556397
dtype: float64

## Since the z score is greater than 2 standard deviation from the mean thus surpassing the threshold, we can reject the null hypothesis.